catalog/spatial/registry/registry-api-impl/src/main/java/org/codice/ddf/registry/api/impl/RegistryStorePublisher.java

```diff
@@ -0,0 +1,201 @@
+import org.codice.ddf.security.common.Security;
+import org.osgi.framework.Bundle;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.Constants;
+import org.osgi.framework.FrameworkUtil;
+import org.osgi.framework.ServiceEvent;
+import org.osgi.framework.ServiceReference;
+import org.osgi.service.event.Event;
+import org.osgi.service.event.EventConstants;
+import org.osgi.service.event.EventHandler;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import ddf.catalog.data.Metacard;
+import ddf.catalog.source.SourceMonitor;
+
+/**
+ * The RegistryStorePublisher controls the auto publication of RegistryStores. Upon creation,
+ * the reference listener will call the bindRegistryStore method and publish if isAutoPush is
+ * true. After editing the RegistryStore handleEvent will evaluate if the publish or unpublish
+ * should occur.
+ */
+public class RegistryStorePublisher implements EventHandler {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(RegistryStorePublisher.class);
+
+    private FederationAdminService federationAdminService;
+
+    private RegistryPublicationService registryPublicationService;
+
+    private ConcurrentHashMap<String, Boolean> registryStoreMap = new ConcurrentHashMap<>();
+
+    public void setFederationAdminService(FederationAdminService federationAdminService) {
+        this.federationAdminService = federationAdminService;
+    }
+
+    public void setRegistryPublicationService(
+            RegistryPublicationService registryPublicationService) {
+        this.registryPublicationService = registryPublicationService;
+    }
+
+    public void bindRegistryStore(ServiceReference reference) {
+        BundleContext bundleContext = getBundleContext();
+
+        if (reference == null || bundleContext == null || bundleContext.getService(reference) == null) {
+            return;
+        }
+
+        RegistryStore registryStore = (RegistryStore) bundleContext.getService(reference);
+
+        Boolean previousState = registryStoreMap.put(registryStore.getConfigurationPid(),
+                registryStore.isAutoPush());
+
+        SourceMonitor storeRegisteredSourceMonitor = new SourceMonitor() {
+            @Override
+            public void setAvailable() {
+                if (registryStore.isAutoPush()) {
+                    registryPublish(registryStore);
+                }
+            }
+
+            @Override
+            public void setUnavailable() {
+            }
+        };
+
+        if (registryStore.isAvailable(storeRegisteredSourceMonitor) && registryStore.isAutoPush()) {
+            if (previousState == null || !previousState) {
+                registryPublish(registryStore);
+            }
+        }
+    }
+
+    public void unbindRegistryStore(ServiceReference reference) {
+
+        BundleContext bundleContext = getBundleContext();
+
+        if (reference != null && bundleContext != null) {
```

Comment:
Could this be `RegistryStore registryStore = bundleContext.getService(reference)` if the above comment works?